vissl/losses/barlow_twins_loss.py

```
@@ -0,0 +1,124 @@
+
+        self.loss_config = loss_config
+        self.bt_criterion = BarlowTwinsCriterion(
+            self.loss_config.lambda_, self.loss_config.embedding_dim
+        )
+
+    @classmethod
+    def from_config(cls, loss_config: AttrDict):
+        """
+        Instantiates BarlowTwinsLoss from configuration.
+
+        Args:
+            loss_config: configuration for the loss
+
+        Returns:
+            BarlowTwinsLoss instance.
+        """
+        return cls(loss_config)
+
+    def forward(self, output, target):
+        loss = self.bt_criterion(output)
+        return loss
+
+    def __repr__(self):
+        repr_dict = {
+            "name": self._get_name(),
+            "lambda_": self.loss_config.lambda_,
+            "embedding_dim": self.loss_config.embedding_dim
+        }
+        return pprint.pformat(repr_dict, indent=2)
+
+
+class BarlowTwinsCriterion(nn.Module):
+    """
+    The criterion corresponding to the Barlow Twins loss as defined in the paper
+    https://arxiv.org/abs/2103.03230v1.
+
+    Args:
+        lambda_ (float): weight on the off-diagonal terms. It controls the trade-off between
+            the importance given to the invariance term versus the redundancy reduction term.
+        embedding_dim (int): dimensionality of the representation
+    """
+
+    def __init__(self, lambda_: float, embedding_dim: int):
+        super(BarlowTwinsCriterion, self).__init__()
+
+        self.use_gpu = get_cuda_device_index() > -1
+        self.lambda_ = lambda_
+        self.embedding_dim = embedding_dim
+        self.num_copies = 2
+
+        identity_matrix = torch.eye(embedding_dim)
+        if self.use_gpu:
+            identity_matrix = identity_matrix.cuda(non_blocking=True)
+
+        self.identity_matrix = identity_matrix
+
+    def forward(self, embedding: torch.Tensor):
+        """
+        Calculate the loss. Operates on embeddings tensor.
+        """
+        assert embedding.ndim == 2
```

let's throw an assert error message 
`assert embedding.ndim == 2, f"Incorrect embedding dimension: {embedding.ndim} but expected 2"`

---

Ok. I will update this line and the one in the SimCLR criterion.